packages/next/index.ts

```diff
@@ -0,0 +1,66 @@
+import { offsetFromRoot, readWorkspaceJson } from '@nrwl/workspace';
+import { appRootPath } from '@nrwl/workspace/src/utils/app-root';
+import next from 'next';
+import * as path from 'path';
+import TsConfigPathsPlugin from 'tsconfig-paths-webpack-plugin';
+import { NextBuildBuilderOptions } from './src/builders/dev-server/dev-server.impl';
+
+type NextServer = ReturnType<typeof next>;
+
+/**
+ * If a `customServerTarget` is specified, it's assumed to export a `StartServerFn` function.
+ * Nx will call this function when dev-server command is run. This function should call
+ * `nextApp.prepare()`, and then start an HTTP server on the given `settings.port`.
+ */
+export type StartServerFn = (
+  nextApp: NextServer,
+  settings: NextBuildBuilderOptions
+) => Promise<void>;
+
+/**
+ * A Next.js configuration enhancer which must be used to ensure Next.js output is written to
+ * the correct directory, and to ensure TypeScript sources are compiled when imported from other
+ * packages in the workspace.
+ *
+ * @example
+ *   // myapp/next.config.js
+ *   const { withNx } = require('@nrwl/next');
+ *   module.exports = withNx('myapp')({});
+ */
+const withNx = (projectName: string) => (nextConfig: any = {}) => {
```

Comment:
Not a big deal, but I think we usually expect `index.ts` files to be basically barrel files. You could move this to it's own file and re-export here.

It might also be a good idea to add some unit tests for this?

Reply:
Good idea, I'll add some tests!